src/main/java/org/snomed/snowstorm/rest/ConceptController.java

```diff
@@ -495,6 +500,25 @@ public AsyncConceptChangeBatch getConceptBulkChange(@PathVariable String branch,
 
 		timer.checkpoint("Find children");
 
+		// For each child, determine if any its descendants are members of the passed-in
+		// refset
+		if (!refsetId.equals("")) {
+			// Calculate refset membership once, and pass in as a filter for each
+			// child-descendents query
+			Collection<Long> refsetMemberIds = eclQueryService.selectConceptIds("^" + refsetId,
+					versionControlHelper.getBranchCriteria(branch), branch, true, null, null).getContent();
+
+			for (ConceptMini child : children) {
+
+				if (eclQueryService.hasAnyResults("<< " + child.getConceptId(), branch,
+						versionControlHelper.getBranchCriteria(branch), true, refsetMemberIds)) {
+					child.addExtraField("descendantsAreMemberOfRefset", "true");
```

Comment:
Using a boolean value here, rather than a String, would be more consistent with the rest of the API.

Also you could avoid writing `child.addExtraField("descendantsAreMemberOfRefset",` on two separate lines by taking the boolean result of the `hasAnyResults` call and using that as the value.

Reply:
Agreed on both counts - implemented.